lib/elixir/lib/record.ex

```diff
@@ -36,6 +36,10 @@ defmodule Record do
 
   """
 
+  @type type :: :defrecord | :defrecordp
+  @type name :: atom
+  @type tag :: atom
```

Comment:
FWIW maybe this is the case with `name` as well. We can use `name :: atom` in `extract` maybe and push off the need for introducing the public type. It's easier to add stuff later than remove it :)